torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -38,16 +38,14 @@ def horizontal_flip_bounding_box(
 
     # TODO: Investigate if it makes sense from a performance perspective to have an implementation for every
     #  BoundingBoxFormat instead of converting back and forth
-    bounding_box = (
-        bounding_box.clone()
-        if format == features.BoundingBoxFormat.XYXY
-        else convert_format_bounding_box(bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY)
+    bounding_box = convert_format_bounding_box(
+        bounding_box.clone(), old_format=format, new_format=features.BoundingBoxFormat.XYXY, inplace=True
```

Comment:
If the kernel does internally an inplace operation, then we clone first and pass inplace=True. This ensures the data won't be modified.

Reply:
I think the idiom clone + inplace is slower than to just perform the regular operation in the first place. Meaning, this patch might mean a perf hit for this kernel for XYXY boxes. Have you benchmarked it? Unless this actually significant, I'm happy to keep your patch because it is much cleaner.

Reply:
Previously we did clone outside of the invocation if already xyxy. Otherwise we were producing new tensors in every actual modification within `convert_format_bounding_box`. This one theoretically just does clone and inplace ops everywhere else. At least that's the intention but I'm seeing it that some inplace ops might be harder to do on some conversions (looking into it right now). Do you agree that what I say in regards to clone + inplace is identical, provided that all internal transforms are indeed inplace and don't copy?

Reply:
Perfect thanks for confirming the question on the clone. You are right about CXCYWH it was a mess to right properly but I have a fast version that seems compatible. Let me know your thoughts.